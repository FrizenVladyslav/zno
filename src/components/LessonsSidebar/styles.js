import { StyleSheet } from 'aphrodite/no-important'

export default StyleSheet.create({
  sidebar: {
    padding: 10,
    minHeight: '50vh',
    maxHeight: '100vh',
    marginLeft: 10,
  },
})
