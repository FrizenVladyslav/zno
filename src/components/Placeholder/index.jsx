import React from 'react'
import PropTypes from 'prop-types'
import { Placeholder } from 'semantic-ui-react'

const PlaceholderGenerator = ({ paragraphs }) => {
  return (
    <Placeholder>
      {paragraphs.map((line, index) => (
        <Placeholder.Paragraph key={index}>
          {[...Array(line)].map((l, index) => (
            <Placeholder.Line key={index} />
          ))}
        </Placeholder.Paragraph>
      ))}
    </Placeholder>
  )
}

PlaceholderGenerator.propTypes = {
  paragraphs: PropTypes.arrayOf(PropTypes.number),
}

export default PlaceholderGenerator
