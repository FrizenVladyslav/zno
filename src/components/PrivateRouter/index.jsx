import React from 'react'
import { Route, Redirect } from 'react-router'
import { toast } from 'react-toastify'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

const PrivateRoute = props => {
  let allow = false

  // here private logic ...
  allow = props.roles.includes(props.role)

  // show a notification
  if (!allow) toast.info(props.message || `You have not access to this page`)

  return allow ? (
    <Route {...props} />
  ) : (
    <Redirect to={props.redirectTo ? props.redirectTo : '/'} />
  )
}

PrivateRoute.propTypes = {
  roles: PropTypes.arrayOf(PropTypes.string).isRequired,
  message: PropTypes.string,
  redirectTo: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
}

// @ts-ignore
export default connect(({ user }) => ({ role: user.role }))(PrivateRoute)
