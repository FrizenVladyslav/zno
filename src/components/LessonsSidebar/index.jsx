import React from 'react'
import { Menu, Segment } from 'semantic-ui-react'
import { css } from 'aphrodite'

import styles from './styles'

const Sidebar = () => {
  return (
    <Menu vertical className={css(styles.sidebar)}>
      <Menu.Item header>Предмет</Menu.Item>
      <Menu.Item name="Українська мова" />
    </Menu>
  )
}

export default Sidebar
