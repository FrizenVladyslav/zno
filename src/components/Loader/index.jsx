import React from 'react'
import PropTypes from 'prop-types'
import { Loader as SemanticLoader } from 'semantic-ui-react'
import { css } from 'aphrodite/no-important'

import styles from './styles'

const Loader = props => {
  return (
    <div
      className={css(styles.loaderContainer)}
      style={{ minHeight: props.height }}
    >
      <SemanticLoader active inline="centered" />
    </div>
  )
}

Loader.propTypes = {
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
}

export default Loader
