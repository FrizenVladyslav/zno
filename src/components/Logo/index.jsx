import React from 'react'

import { Link } from 'react-router-dom'
import { Icon } from 'semantic-ui-react'
import { css } from 'aphrodite/no-important'
import styles from './styles'

const Logo = props => {
  return (
    <Link to="/" className={css(styles.logo)}>
      <Icon name="student" size="huge" />
      <h2>Краще за всіх</h2>
    </Link>
  )
}

export default Logo
