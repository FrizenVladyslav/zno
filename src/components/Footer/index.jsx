import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { Container, Segment } from 'semantic-ui-react'
import { css } from 'aphrodite/no-important'
import Logo from '../Logo'

import styles from './styles'

const Footer = props => {
  return (
    <Segment placeholder className={css(styles.footer)}>
      <Container>
        <div>
          <Logo style={{ alignSelf: 'flex-end' }} />
          <p>&copy; Всі права захищені)</p>
        </div>
        <ul>
          <li>Предмети:</li>
          {props.lessons &&
            props.lessons.map(lesson => (
              <li key={lesson._id}>
                <Link to={`/lessons/${lesson._id}`}>{lesson.title}</Link>
              </li>
            ))}
        </ul>
        <ul>
          <li>Інші посилання:</li>
          <li>
            <Link to="/login">Увійти</Link>
          </li>
          <li>
            <Link to="/lessons">Зайняття</Link>
          </li>
        </ul>
      </Container>
    </Segment>
  )
}

// @ts-ignore
export default connect(({ lesson }) => ({ lessons: lesson.lessons }))(Footer)
