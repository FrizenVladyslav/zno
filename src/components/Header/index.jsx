import React, { useState } from 'react'
import { connect } from 'react-redux'
import { Menu } from 'semantic-ui-react'
import { NavLink } from 'react-router-dom'
import { css } from 'aphrodite'
import Logo from '../Logo'
import ConfirmModal from '../Modals/ConfirmModal'

import * as userActions from 'actions/user'

import styles from './styles'

const Header = props => {
  const [modal, triggerModal] = useState(false)

  return (
    <>
      <Menu stackable className={css(styles.menu)}>
        <Menu.Header>
          <Logo />
        </Menu.Header>
        <Menu.Menu position="right">
          {props.user.role === 'admin' && (
            <Menu.Item name="admin" link as={NavLink} to="/admin">
              Адміністрування
            </Menu.Item>
          )}
          <Menu.Item name="lessons" link as={NavLink} to="/lessons">
            Предмети
          </Menu.Item>
          {!!props.user._id ? (
            <Menu.Item link onClick={() => triggerModal(true)}>
              {props.user.nick}
            </Menu.Item>
          ) : (
            <Menu.Item as={NavLink} link name="sign-in" to="/login">
              Увійти
            </Menu.Item>
          )}
        </Menu.Menu>
      </Menu>
      <ConfirmModal
        icon="sign-out"
        message="Ви дійсно бажаєте вийти?"
        open={modal}
        onClose={() => triggerModal(false)}
        onSuccess={userActions.signOut}
        title="Вийти"
      />
    </>
  )
}

// @ts-ignore
export default connect(({ user }) => ({ user }))(Header)
