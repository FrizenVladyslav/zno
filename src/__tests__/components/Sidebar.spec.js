// @ts-nocheck
import React from 'react'
import { shallow } from 'enzyme'
import { Sidebar } from 'components/LessonsSidebar'
import { StyleSheetTestUtils } from 'aphrodite'

describe('sidebar', () => {
  afterEach(() => {
    return new Promise(resolve => {
      StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
      return process.nextTick(resolve)
    })
  })

  const lessons = [
    {
      _id: '1',
      title: 'title1',
    },
    {
      _id: '2',
      title: 'title2',
    },
  ]
  const sidebar = shallow(<Sidebar activeLesson={'2'} lessons={lessons} />)

  describe('without lessons', () => {
    const sidebar = shallow(<Sidebar activeLesson={null} lessons={[]} />)

    it('render properly', () => {
      expect(sidebar).toMatchSnapshot()
    })

    it('not have excess items', () => {
      expect(sidebar.find('MenuItem')).toHaveLength(1)
    })
  })

  describe('with lessons', () => {
    it('render properly', () => {
      expect(sidebar).toMatchSnapshot()
    })

    it('renders all items', () => {
      expect(sidebar.find('MenuItem')).toHaveLength(3)
    })

    describe('item render correctly', () => {
      const propsItem = sidebar
        .find('MenuItem')
        .last()
        .props()

      it('name', () => {
        expect(propsItem.name).toBe(lessons[1].title)
      })

      it('link path', () => {
        expect(propsItem.to).toBe(`/lessons/${lessons[1]._id}`)
      })

      it('item active', () => {
        expect(propsItem.active).toBeTruthy()
      })
    })
  })
})
