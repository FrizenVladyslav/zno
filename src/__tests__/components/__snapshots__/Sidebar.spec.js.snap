// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar with lessons render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar
    activeLesson="2"
    lessons={
      Array [
        Object {
          "_id": "1",
          "title": "title1",
        },
        Object {
          "_id": "2",
          "title": "title2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <MenuItem
          header={true}
        >
          Предмет
        </MenuItem>,
        Array [
          <MenuItem
            active={false}
            as={[Function]}
            name="title1"
            to="/lessons/1"
          />,
          <MenuItem
            active={true}
            as={[Function]}
            name="title2"
            to="/lessons/2"
          />,
        ],
      ],
      "className": "sidebar_amlubi",
      "pointing": true,
      "vertical": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Предмет",
          "header": true,
        },
        "ref": null,
        "rendered": "Предмет",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "active": false,
          "as": [Function],
          "name": "title1",
          "to": "/lessons/1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "active": true,
          "as": [Function],
          "name": "title2",
          "to": "/lessons/2",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <MenuItem
            header={true}
          >
            Предмет
          </MenuItem>,
          Array [
            <MenuItem
              active={false}
              as={[Function]}
              name="title1"
              to="/lessons/1"
            />,
            <MenuItem
              active={true}
              as={[Function]}
              name="title2"
              to="/lessons/2"
            />,
          ],
        ],
        "className": "sidebar_amlubi",
        "pointing": true,
        "vertical": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Предмет",
            "header": true,
          },
          "ref": null,
          "rendered": "Предмет",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "active": false,
            "as": [Function],
            "name": "title1",
            "to": "/lessons/1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "active": true,
            "as": [Function],
            "name": "title2",
            "to": "/lessons/2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`sidebar without lessons render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar
    activeLesson={null}
    lessons={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <MenuItem
          header={true}
        >
          Предмет
        </MenuItem>,
        Array [],
      ],
      "className": "sidebar_amlubi",
      "pointing": true,
      "vertical": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Предмет",
          "header": true,
        },
        "ref": null,
        "rendered": "Предмет",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <MenuItem
            header={true}
          >
            Предмет
          </MenuItem>,
          Array [],
        ],
        "className": "sidebar_amlubi",
        "pointing": true,
        "vertical": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Предмет",
            "header": true,
          },
          "ref": null,
          "rendered": "Предмет",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
